services/transcode/playout_mux.go

```diff
@@ -0,0 +1,247 @@
+package transcode
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bcc-code/bccm-flows/common"
+	"github.com/bcc-code/bccm-flows/services/ffmpeg"
+)
+
+/***
+
+# Playout mux
+
+The playout system has these known requirements:
+- Exactly 16 audio tracks representing 12 languages
+- The audio tracks are all mono
+- The first 8 audio tracks are stereo pairs (L and R) for the 4 first languages
+- The next 8 audio tracks are mono for the 8 next languages
+
+It's not known if this is a requirement, but we also:
+- resample audio to 48khz (ffmpeg anyway cant use 44.1khz for pcm? got an error)
+- reencode audio to 24bit pcm_s24le
+
+## Example ffmpeg command
+The following is a simpler version, to make it easier to understand the filter.
+
+- Resample all audio to 48khz (aresample=48000)
+- It creates 5 audio tracks.
+  - Track 1 and 2 will be norwegian L and R (channelsplit=channel_layout=stereo)
+  - Track 3 and 4 will be english L and R. (channelsplit=channel_layout=stereo)
+  - Track 5 will be finnish L (pan=1c|c0=c0)
+  - Track 6 will be norwegian L (the norwegian L is split into two streams: `asplit=2[nor_l_copy1][nor_l_copy2]`)
+- Subtitles are not included.
+
+```bash
+
+ffmpeg \
+  -i BERG_TS01_ISRAEL_VOD.mxf \
+  -i BERG_TS01_ISRAEL_VOD-nor.wav \
+  -i BERG_TS01_ISRAEL_VOD-eng.wav \
+  -i BERG_TS01_ISRAEL_VOD-fin.wav \
+  -filter_complex "[1:a]aresample=48000,channelsplit=channel_layout=stereo[nor_l][nor_r]; [nor_l]asplit=2[nor_l_copy1][nor_l_copy2]; [2:a]aresample=48000,channelsplit=channel_layout=stereo[eng_l][eng_r]; [3:a]aresample=48000,pan=1c|c0=c0[fin_l]" \
+  -map "0:v" -map "[nor_l_copy1]" -map "[nor_r]" -map "[eng_l]" -map "[eng_r]" -map "[fin_l]" -map "[nor_l_copy2]" \
+  -c:v copy -c:a pcm_s24le \
+  -y transcoded/output.mxf
+
+```
+
+**/
+
+var playoutLanguages = [12]string{
+	"nor",
+	"deu",
+	"nld",
+	"eng",
+	"fra",
+	"spa",
+	"fin",
+	"rus",
+	"por",
+	"ron",
+	"tur",
+	"pol",
+}
+
+func createStereoFilter(input string, leftOutput string, rightOutput string) string {
+	return fmt.Sprintf("[%s]aresample=48000,channelsplit=channel_layout=stereo[%s][%s]", input, leftOutput, rightOutput)
+}
+
+func createMonoFilter(input string, output string) string {
+	return fmt.Sprintf("[%s]aresample=48000[%s]", input, output)
+}
+
```

Comment:
This requires a reencode of the audio streams. Might want to split it into a separate step? Or is this the only place where this audio is used?

Reply:
Yea this is very specific to playout..
On the "muxing is free" bit: I've been kind of unsure about calling it "mux" because it kinda sounds like just moving streams around (free), but I went for mux anyway because I havent seen any definition that says multiplexing is free, it just means packing stuff into one file

Reply:
The definition of multiplexing is combining separate sources into one medium, mostly lossless.

It's not important if we don't confuse it.